src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -46,16 +46,33 @@ public bool TryRegister(SafeCloseSocket socket, Interop.Sys.SocketEvents current
             }
         }
 
+#if DEBUG
         private const int EventBufferCount = 64;
+#else
+        private const int EventBufferCount = 1024;
+#endif
 
         private static readonly object s_lock = new object();
 
+#if DEBUG
+        //
+        // In debug builds, force there to number of engines to 2
+        //
+        private static readonly int EngineCount = 2;
+#else
+        //
+        // In release builds, use half the number of processors when there are at least 6 processors.
+        // The lower bound is to avoid spinning up many engines on systems which aren't servers.
         //
-        // The current engine.  We replace this with a new engine when we run out of "handle" values for the current
+        private static readonly int EngineCount = Environment.ProcessorCount >= 6 ? Environment.ProcessorCount >> 1 : 1;
+#endif
+        //
+        // The current engines.  We replace this an engine when we run out of "handle" values for the current
```

Comment:
Nit: "replace this an engine when we run out of "handle" values for the current engine" => "replace these engines when we run out of "handle" values for the current ones"